components/event-simulator/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/ds/CarbonEventSimulator.java

```diff
@@ -926,6 +944,7 @@ public void run(){
                     }
                     event.setAttributeValues(attributeValues);
                     sendEvent(event);
+                    Thread.sleep(delayBetweenEventsInMilies);
```

Comment:
Thanks @lasanthafdo, If the delay is 0 or negative, it will do send a warning log and send events continuously. Changes are done in pull request - https://github.com/wso2/carbon-event-processing/pull/160